pulp_container/app/registry_api.py

```diff
@@ -979,12 +976,20 @@ def handle_safe_method(self, request, path, pk):
         except models.Blob.DoesNotExist:
             if pk == EMPTY_BLOB:
                 return redirects.redirect_to_content_app("blobs", pk)
-            repository = repository.cast()
             try:
-                blob = repository.pending_blobs.get(digest=pk)
-                blob.touch()
-            except models.Blob.DoesNotExist:
-                raise BlobNotFound(digest=pk)
+                blob = models.ConfigBlob.objects.get(digest=pk)
+                return redirects.redirect_to_content_app("config-blobs", pk)
+            except models.ConfigBlob.DoesNotExist:
+                repository = repository.cast()
+                try:
+                    blob = repository.pending_blobs.get(digest=pk)
+                    blob.touch()
+                except models.Blob.DoesNotExist:
+                    try:
+                        blob = repository.pending_config_blobs.get(digest=pk)
+                        blob.touch()
+                    except models.ConfigBlob.DoesNotExist:
+                        raise BlobNotFound(digest=pk)
```

Comment:
If we could have the relation between the Blob and ConfigBlob model, we could reduce the complexity a bit.

Reply:
Maybe it would make sense to revisit a concept of having a single blob model which has a ".data" field, containing the extracted file data in case of a config blob.